docs/architecture.md

```diff
@@ -0,0 +1,54 @@
+# Architecture
+
+The language features provided by the [MicroProfile Language Server](./microprofile.ls) depend on the dependencies in the user's Java Project.
+For example, MicroProfile Fault Tolerance features (snippets, validation, completion, etc.) are only provided if the user's
+Java project has MicroProfile Fault Tolerance as a dependency.
+
+In order to avoid duplicating the work done by an IDEs Java Language Support (computing the classpath, rebuilding AST, etc.) the MicroProfile language Server relies on an external component that understands and handles LSP command requests sent from the MicroProfile Language Server.
+
+The MicroProfile Language Server has limited knowledge of the users Java project, and instead uses the external component for information on the Java project. In the context of Eclipse and VS Code, this external component is managed with [MicroProfile JDT](./microprofile.jdt).
+The external component does not necessarily require the JDT however, as different implementations can be used instead. For example [Quarkus Tools for IntelliJ](https://github.com/redhat-developer/intellij-quarkus) does not use the JDT and instead has its own implementation used to handle the requests from [`MicroProfileLanguageClientAPI`](https://github.com/eclipse/lsp4mp/blob/master/microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/ls/api/MicroProfileLanguageClientAPI.java). In other words, the MicroProfile Language Server **does not** depend on JDT.
```

Comment:
@jeffmaury could you review this sentence please about IJ, if you think we should add another content. perhaps it should be fine to explain that you are using IJ Psi java API (I'm not sure if it's the right name).

Reply:
You can rewrite:
`does not use the JDT and instead has its own implementation used to handle the requests`
as
`does not use the JDT and instead has its own implementation based on [Program Structure Interface](https://jetbrains.org/intellij/sdk/docs/basics/architectural_overview/psi.html) used to handle the requests`

Reply:
Thanks @jeffmaury!